sympy/core/numbers.py

```diff
@@ -4436,8 +4436,107 @@ def as_base_exp(self):
     def _mpc_(self):
         return (Float(0)._mpf_, Float(1)._mpf_)
 
+
 I = S.ImaginaryUnit
 
+
+def equal_valued(x, y):
+    """Compare expressions treating plain floats as rationals.
+
+    Examples
+    ========
+
+    >>> from sympy import S, symbols, Rational, Float
+    >>> from sympy.core.numbers import equal_valued
+    >>> equal_valued(1, 2)
+    False
+    >>> equal_valued(1, 1)
+    True
+
+    In SymPy expressions with Floats compare unequal to corresponding
+    expressions with rationals:
+
+    >>> x = symbols('x')
+    >>> x**2 == x**2.0
+    False
+
+    However an individual Float compares equal to a Rational:
+
+    >>> Rational(1, 2) == Float(0.5)
+    True
+
+    In a future version of SymPy this might change so that Rational and Float
+    compare unequal. This function provides the behaviour currently expected of
+    ``==`` so that it could still be used if the behaviour of ``==`` were to
```

Comment:
```suggestion
    compare unequal. This function provides the behavior currently expected of
    ``==`` so that it could still be used if the behavior of ``==`` were to
```